src/main/java/io/pinecone/clients/Inference.java

```diff
@@ -0,0 +1,40 @@
+package io.pinecone.clients;
+
+import org.openapitools.control.client.ApiException;
+import org.openapitools.control.client.api.InferenceApi;
+import org.openapitools.control.client.model.EmbedRequest;
+import org.openapitools.control.client.model.EmbedRequestInputsInner;
+import org.openapitools.control.client.model.EmbedRequestParameters;
+import org.openapitools.control.client.model.EmbeddingsList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Inference {
+    InferenceApi inferenceApi;
+
+    public Inference() {
+        inferenceApi = new InferenceApi();
+    }
+
+    public EmbeddingsList embed(String model, String truncate, String inputType, List<String> inputs) throws ApiException {
+        EmbedRequestParameters embedRequestParameters = new EmbedRequestParameters().inputType(inputType);
+        if(truncate != null && !truncate.isEmpty())
+            embedRequestParameters.truncate(truncate);
+        List<EmbedRequestInputsInner> EmbedRequestInputsInnerList = convertInputStringToEmbedRequestInputsInner(inputs);
+        EmbedRequest embedRequest = new EmbedRequest()
+                .model(model)
+                .parameters(embedRequestParameters)
+                .inputs(EmbedRequestInputsInnerList);
```

Comment:
Other models might support different so would be better as a map.

Reply:
Can you please elaborate on this more? Are you suggesting `List<String> inputs` should be a `map` instead? If so, how does the key value pair be assigned to the `inputs` since `inputs` is of type `List<EmbedRequestInputsInner>` where `EmbedRequestInputsInner` has only one member which is `text` of type String.

Reply:
I mean for the params like truncate and input_type -- those apply to the multilingual-e5-large model, but they might not apply to the next embedding model we may add. 

You can use the Python impl as a reference, where it takes params as basically an optional map (I know, Java optionals are not a thing)